component/src/main/java/org/ballerinalang/net/kafka/nativeimpl/functions/consumer/SubscribeWithPartitionRebalance.java

```diff
@@ -0,0 +1,187 @@
+
+        private Context context;
+        private FunctionRefCPEntry onPartitionsRevoked;
+        private FunctionRefCPEntry onPartitionsAssigned;
+        private AbstractNativeFunction function;
+        private BStruct consumerStruct;
+
+
+        KafkaRebalanceListener(Context context,
+                               FunctionRefCPEntry onPartitionsRevoked,
+                               FunctionRefCPEntry onPartitionsAssigned,
+                               AbstractNativeFunction function,
+                               BStruct consumerStruct) {
+            this.context = context;
+            this.onPartitionsRevoked = onPartitionsRevoked;
+            this.onPartitionsAssigned = onPartitionsAssigned;
+            this.function = function;
+            this.consumerStruct = consumerStruct;
+        }
+
+        @Override
+        public void onPartitionsRevoked(Collection<TopicPartition> partitions) {
+            ProgramFile programFile = context.getProgramFile();
+            Context childContext = new WorkerContext(programFile, context);
+            BLangFunctions.
+                    invokeFunction(programFile, onPartitionsRevoked.getFunctionInfo(),
+                            function.getBValues(consumerStruct, getPartitionsArray(partitions)), childContext);
+        }
+
+        @Override
+        public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
+            ProgramFile programFile = context.getProgramFile();
+            Context childContext = new WorkerContext(programFile, context);
+            BLangFunctions.
+                    invokeFunction(programFile, onPartitionsAssigned.getFunctionInfo(),
+                            function.getBValues(consumerStruct, getPartitionsArray(partitions)), childContext);
+
+        }
+
+        private BRefValueArray getPartitionsArray(Collection<TopicPartition> partitions) {
+            List<BStruct> assignmentList = new ArrayList<>();
+            if (!partitions.isEmpty()) {
+                partitions.forEach(assignment -> {
+                    BStruct infoStruct = createPartitionStruct(context);
+                    infoStruct.setStringField(0, assignment.topic());
+                    infoStruct.setIntField(0, assignment.partition());
+                    assignmentList.add(infoStruct);
+                });
+            }
+            return new BRefValueArray(assignmentList.toArray(new BRefType[0]),
+                    createPartitionStruct(context).getType());
+        }
+
+        private BStruct createPartitionStruct(Context context) {
+            PackageInfo kafkaPackageInfo = context.getProgramFile()
+                    .getPackageInfo(Constants.KAFKA_NATIVE_PACKAGE);
+            StructInfo consumerRecordStructInfo = kafkaPackageInfo
+                    .getStructInfo(Constants.TOPIC_PARTITION_STRUCT_NAME);
+            BStructType structType = consumerRecordStructInfo.getType();
+            BStruct bStruct = new BStruct(structType);
```

Comment:
Local variable bstruct is redundant here. We can use `return new BStruct(structType);`

Reply:
+1